app/src/main/java/com/haruta/harutyan/countchallenge/MainActivity.kt

```diff
@@ -14,7 +15,16 @@ class MainActivity : AppCompatActivity() {
 
         binding.tapAddButton.setOnClickListener {
             countNumber += 1
+            changeTextColor()
             binding.viewNumberTextview.text = countNumber.toString()
         }
     }
+
+    fun changeTextColor() {
```

Comment:
せっかくなので少し難しい話を紹介します！

実は関数は、引数を受け取って返り値を返す方が良いとされています。
その理由の一つは、どんな値を渡してどんな結果が返ってくるかが分かることで、関数内でどのような処理がされるか分かりやすいことです。

逆に引数や戻り値がないと、関数の中で何をしているのかが想像しにくいです。
例えば View の状態などを予期せぬように変えてしまったりしても、気づきにくいです。
これを副作用と呼んだりします。

今回の関数は countNumber を受け取ってカラーコードを返す関数とも考えられるので、以下のように書き換えることもできます！

```suggestion
  override fun onCreate(savedInstanceState: Bundle?) {
    // 省略
    binding.tapAddButton.setOnClickListener {
      countNumber += 1
      val colorCode = getColorCode(countNumber)
      binding.viewNumberTextview.setTextColor(Color.parseColor(colorCode))
      binding.viewNumberTextview.text = countNumber.toString()
    }
  }

    fun getColorCode(countNumber: Int): String {
      return when(countNumber % 2) {
        0 -> "#0000FF"
        1 -> "#FF0000"
        else -> "#000000"
      }
    }
```